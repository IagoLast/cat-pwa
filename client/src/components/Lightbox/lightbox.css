.lightbox {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: 0;
  padding: 38px;
  background: rgba(0, 0, 0, 0.7);
  animation: fadeIn 1s;
  animation-fill-mode: forwards;
}

.lightbox__figure {
  position: relative;
  background: white;
  margin: auto;
  height: 100%;
  max-width: 400px;
  overflow: auto;
  border-radius: 2px;
}

.lightbox__img {
  width: 100%;
  height: auto;
  background: #f2f2f2;
}

.lightbox__caption {
}

.lightbox__caption:hover {
}


@keyframes fadeIn {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}