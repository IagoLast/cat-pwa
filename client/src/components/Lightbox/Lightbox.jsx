import React, { Component } from 'react';
import './lightbox.css';

export default class Lightbox extends Component {
  render() {
    return <div className="lightbox" onClick={this.props.handleClick}>
      <figure className="lightbox__figure">
        <img className="lightbox__img"
          width={this.props.data.width_l}
          height={this.props.data.height_l}
          data-src={this.props.data.url_l}
          alt={this.props.data.title}
          title={this.props.data.title} />
        <figcaption className="lightbox__caption">
          <p className="date">{this.props.data.datetaken}</p>
          <h1 className="title"> {this.props.data.title ||'Picture'} </h1>
          <h3 className="owner"> @{this.props.data.ownername}</h3>
          <p className="tags">{this.props.data.tags}</p>
        </figcaption>
      </figure>
    </div>
  }
}
