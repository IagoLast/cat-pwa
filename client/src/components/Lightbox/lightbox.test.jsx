import React from 'react';
import renderer from 'react-test-renderer';

import Lightbox from './Lightbox';

it('renders correctly', () => {
  const data = {
    datetaken: 'fake-date-taken',
    url_l: 'fake-url',
    title: 'fake-title',
    ownername: 'fake-ownername',
    ownername: 'fake-tags_0, fake-tags_1',
  }
  const tree = renderer.create(<Lightbox handleClick={console.log} data={data}></Lightbox>).toJSON();
  expect(tree).toMatchSnapshot();
});