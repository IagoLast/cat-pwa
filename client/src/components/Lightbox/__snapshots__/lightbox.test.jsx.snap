// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="lightbox"
  onClick={[Function]}
>
  <div
    className="lightbox__figure"
  >
    <div
      className="lightbox__img"
      style={
        Object {
          "backgroundImage": "url(\\"fake-url\\")",
        }
      }
    />
  </div>
  <div
    className="lightbox__caption"
  >
    <p
      className="date"
    >
      fake-date-taken
    </p>
    <h1
      className="title"
    >
       
      fake-title
       
    </h1>
    <h3
      className="owner"
    >
       @
      fake-tags_0, fake-tags_1
    </h3>
    <p
      className="tags"
    />
  </div>
</div>
`;
